airbyte_cdk/sources/declarative/concurrent_declarative_source.py

```diff
@@ -107,8 +108,8 @@ def __init__(
                 concurrency_level // 2, 1
```

Comment:
Should this also be set to `_LOWEST_SAFE_CONCURRENCY_LEVEL`? Otherwise a customer could accidentally configure this to still use the unsafe 1

Reply:
For now, this would be blocking the only "single threaded mode" that we can have. Even though imperfect, I'm wondering if we should still allow it just in case. When we will see an issue with this, the action item is probably to implement the `single thread mode` mentioned [here](https://github.com/airbytehq/airbyte-python-cdk/pull/148#discussion_r1873889010)